File: R/LR.R
library(caret)
library(readr)

data <- read_delim('datasets/wine_quality.csv', delim = ";", col_names = TRUE)

X <- data[ , !names(data) %in% 'quality']
y <- data$quality

set.seed(42)
trainIndex <- createDataPartition(y, p = .8, list = FALSE)
X_train <- X[trainIndex, ]
X_test <- X[-trainIndex, ]
y_train <- y[trainIndex]
y_test <- y[-trainIndex]

model <- train(
  X_train,
  y_train,
  method = "lm",
  trControl = trainControl(method = "none")
)

File: Python/LR.py
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

data = pd.read_csv('datasets/wine_quality.csv', sep=';')
X = data.drop('quality', axis=1)
y = data['quality']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

model = LinearRegression(fit_intercept=True)

model.fit(X_train, y_train)